### DISCUSSION TITLE
ModuleNotFoundError: No module named 'requests'

### DISCUSSION THREAD

**Discussion by thouyear** [oobabooga/text-generation-webui#3464] [Q&A]
![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/assets/138983583/39ed99d3-5156-4f81-ae49-554da39f7955)

It seems that there is no such requests.py in the source file when i use text-generation-webui

![0628c6d82d021d1caa45011abb18a32](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/assets/138983583/bc081b03-ee21-45c2-9d53-aa4c97d5de86)

**Comment by ossirytk** [comment]
Requests is a python module. I'm guessing that you don't have it installed in the textgen virtual environment. Try running `pip list`.You might want to run `pip install -r requirements.txt` while inside the textgen virtual environment. That or you have something wrong with your installation.

**Reply by Sara681820** [reply]
Hi, I have already installed requests module. The result of pip list command includes: "requests                  2.31.0" also the result of the command: conda list requests is same version:"requests                  2.31.0          py310h06a4308_0". I also tried pip install -r requirements.txt and the result was requirements already installed. The issue is the above error still exist.

**Reply by wooexperte** [reply]
Hello, I have the same problem and haven't found a solution. Have you found a solution? Best regards Harald

**Reply by OmidH** [reply]
Same here

**Reply by tholonia** [reply]
Here is what I did, and it ONLY worked with python 3.11

```
conda create -n llama python=3.11
conda install -n llama conda-libmamba-solver
conda activate llama
conda config --set channel_priority strict
conda config --set solver libmamba
conda install pip
cd ~/src/text-generation-webui
python -m pip install -r ./requirements.txt
```

edit ` modules/block_requests.py`
add these lines before "`import requests`"
```
import sys
sys.path.append("/home/<username>/.conda/envs/llama/lib/python3.11/site-packages")

```
save

`./start_linux.sh`

works!   Well, sorta works, as it breaks all over the place in different sections, for example, in "Models" tab.

```
File "/home/jw/store/src/text-generation-webui/modules/ui_model_menu.py", line 238, in download_model_wrapper


model, branch = downloader.sanitize_model_and_branch_names(repo_id, None)

                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/home/jw/store/src/text-generation-webui/download-model.py", line 39, in sanitize_model_and_branch_names


if model[-1] == '/':

   ~~~~~^^^^
IndexError: string index out of range
```

:/
